components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -357,6 +400,48 @@ func (agent *Smith) Start(callback func(InfringingWorkspace, []PenaltyKind)) {
 	}
 }
 
+func (agent *Smith) cleanupDeadPIDS(ctx context.Context) {
```

Comment:
It's a personal taste thing, but I'd prefer if `cleanupDeadPIDS` were called as Go routine rather than it spawning one itself.
While reading through I was wondering at the callsite of `cleanupDeadPIDS` if we did this only once. Had the callsite been `go cleanupDeadPIDS()` that would have been clearer.

Reply:
I agree!